### PR TITLE
[vpj] VPJ should not close and release resources on its own

### PR DESCRIPTION
**Pull request by nisargthakkar**
## VPJ should not close and release resources on its own

VPJ with D2 fails to send final push job details to controller because VPJ internally closes the D2 client and the subsequent `ControllerClient` request fails. The URL based ControllerClient doesn't have this issue because the `close` method is essentially a no-op.

## How was this PR tested?

GH CI

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java [hunk 1]
```diff
@@ -3421,10 +3417,9 @@ private void killJobAndCleanup(
         LOGGER.info("Offline push job has been killed, topic: {}", topicInfo.topic);
       }
     }
-    close();
```
Comment (nisargthakkar) [inline clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:3424]: This is the crux of the fix

**Comment by nisargthakkar** [comment]
Leaving in draft phase till all release blockers are resolved. This is not high priority

File: clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java (hunk 1 above)
Comment (sushantmane) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:3424]: We should still be calling close() after sending report, right?

Comment (nisargthakkar) [reply clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java:3424]: Calling `close` is the responsibility of the code that creates the `VenicePushJob` object. It seems weird that we would close an object during it's execution

**Review by sushantmane — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**